source/common/config/config_provider_impl.h

```diff
@@ -200,20 +194,17 @@ class ConfigSubscriptionInstanceBase : public Init::Target,
                                  ConfigProviderManagerImplBase& config_provider_manager,
                                  TimeSource& time_source, const SystemTime& last_updated,
                                  const LocalInfo::LocalInfo& local_info)
-      : name_(name), manager_identifier_(manager_identifier),
-        config_provider_manager_(config_provider_manager), time_source_(time_source),
-        last_updated_(last_updated) {
+      : name_(name), init_target_(fmt::format("ConfigSubscriptionInstanceBase {}", name_),
+                                  [this]() { start(); }),
+        manager_identifier_(manager_identifier), config_provider_manager_(config_provider_manager),
+        time_source_(time_source), last_updated_(last_updated) {
     Envoy::Config::Utility::checkLocalInfo(name, local_info);
   }
 
   void setLastUpdated() { last_updated_ = time_source_.systemTime(); }
 
-  void runInitializeCallbackIfAny();
-
 private:
-  void registerInitTarget(Init::Manager& init_manager) {
-    init_manager.registerTarget(*this, fmt::format("ConfigSubscriptionInstanceBase {}", name_));
-  }
+  void registerInitTarget(Init::Manager& init_manager) { init_manager.add(init_target_); }
```

Comment:
In terms of naming, this is a bit of a weird inversion. Should we just make the init target available to the caller in `getSubscription()` and inline this there?